File: parser-test/src/main/java/ru/spbau/kaysin/parser/domain/CompilationUnit.java
package ru.spbau.kaysin.parser.domain;

import java.util.Collection;

/**
 * Class representing java compilation unit.
 */
public class CompilationUnit {

    private final Collection<ru.spbau.kaysin.parser.domain.Class> classes;

    @Override
    public String toString() {
        return "CompilationUnit{" +
            "classes=" + classes +
            '}';
    }

    public CompilationUnit(Collection<ru.spbau.kaysin.parser.domain.Class> classes) {
        this.classes = classes;
    }
}

File: parser-test/src/main/scala/ru/spbau/kaysin/scala/parser/codeEntities/CompilationUnit.scala
package ru.spbau.kaysin.scala.parser.codeEntities


/**
  * Class representing java compilation unit.
  */
class CompilationUnit(val packageDeclaration: PackageDeclaration,
                      val importDeclarations: Set[ImportDeclaration],
                      val classes: Set[Class]) {

  override def toString = s"CompilationUnit($packageDeclaration, $importDeclarations, $classes)"
}